.github/workflows/auto-update-Dockerfiles.yml

```diff
@@ -0,0 +1,192 @@
+          DOCKERFILE_PATH: ${{ env.NET_9_AMD64_Dockerfile }}
+        run: |
+          $version = & "./LambdaRuntimeDockerfiles/get-latest-aspnet-versions.ps1" -MajorVersion "9"
+          if (-not [string]::IsNullOrEmpty($version)) {
+            & "./LambdaRuntimeDockerfiles/update-dockerfile.ps1" -DockerfilePath "${{ env.DOCKERFILE_PATH }}" -NextVersion $version
+          } else {
+            Write-Host "Skipping .NET 9 AMD64 update - No version detected"
+          }
+
+      # Update .NET 9 ARM64 Dockerfile
+      - name: Update .NET 9 ARM64
+        id: update-net9-arm64
+        shell: pwsh
+        env:
+          DOCKERFILE_PATH: ${{ env.NET_9_ARM64_Dockerfile }}
+        run: |
+          $version = & "./LambdaRuntimeDockerfiles/get-latest-aspnet-versions.ps1" -MajorVersion "9"
+          if (-not [string]::IsNullOrEmpty($version)) {
+            & "./LambdaRuntimeDockerfiles/update-dockerfile.ps1" -DockerfilePath "${{ env.DOCKERFILE_PATH }}" -NextVersion $version
+          } else {
+            Write-Host "Skipping .NET 9 ARM64 update - No version detected"
+          }
+
+      # Update .NET 10 AMD64 Dockerfile
+      - name: Update .NET 10 AMD64
+        id: update-net10-amd64
+        shell: pwsh
+        env:
+          DOCKERFILE_PATH: ${{ env.NET_10_AMD64_Dockerfile }}
+        run: |
+          $version = & "./LambdaRuntimeDockerfiles/get-latest-aspnet-versions.ps1" -MajorVersion "10"
+          if (-not [string]::IsNullOrEmpty($version)) {
+            & "./LambdaRuntimeDockerfiles/update-dockerfile.ps1" -DockerfilePath "${{ env.DOCKERFILE_PATH }}" -NextVersion $version
+          } else {
+            Write-Host "Skipping .NET 10 AMD64 update - No version detected"
+          }
+
+      # Update .NET 10 ARM64 Dockerfile
+      - name: Update .NET 10 ARM64
+        id: update-net10-arm64
+        shell: pwsh
+        env:
+          DOCKERFILE_PATH: ${{ env.NET_10_ARM64_Dockerfile }}
+        run: |
+          $version = & "./LambdaRuntimeDockerfiles/get-latest-aspnet-versions.ps1" -MajorVersion "10"
+          if (-not [string]::IsNullOrEmpty($version)) {
+            & "./LambdaRuntimeDockerfiles/update-dockerfile.ps1" -DockerfilePath "${{ env.DOCKERFILE_PATH }}" -NextVersion $version
+          } else {
+            Write-Host "Skipping .NET 10 ARM64 update - No version detected"
+          }
+
+      # Commit changes and create a branch
+      - name: Commit and Push
+        id: commit-push
+        shell: pwsh
+        run: |
+          # Check if there are any changes to commit
+          if (git status --porcelain) {
+            git config --global user.email "github-aws-sdk-dotnet-automation@amazon.com"
+            git config --global user.name "aws-sdk-dotnet-automation"
+            $branch="chore/auto-update-Dockerfiles-daily"
+            
+            # Check if the branch already exists remotely
+            $branchExists = git ls-remote --heads origin $branch
+            if ($branchExists) {
```

Comment:
im not a fan of the branch exists approach. we are not certain that the changes in that branch are what we want them to be. we shouldn't just add more to it. I'd say let's always create a unique branch by appending a timestamp to the name.

Reply:
fixed in e679308